### PR TITLE
2 Loading Screens

### PR DESCRIPTION
**Pull request by griffinclark10**
2 loading.tsx screens, one for each page type + a styles.css for loading animation

### CODE REVIEW

**Review by robinbb — CHANGES_REQUESTED** [review]
After issuing 'just launch-web' (which launches the server and the db, so that the web component can be tested), I see that these additions do not change anything about how the web pages load. Probably the

**Comment by griffinclark10** [comment]
Hey @robinbb, thanks for the feedback! I fixed all of the comments you pointed out but the loading screen does seem to be working on my end 🤔 . This is what I see when I try and access the results page while the data is loading:

Screenshot 2023-10-29 at 2 35 53 PM

Not sure what the problem could be. Let me know if I'm doing something wrong on my end. Also, if theres anything about the style that wasn't what you were hoping for, I'd be happy to make some changes!

File: web/app/loading.tsx [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import './styles.css'
+
+function LoadingDots() {
```
Comment (robinbb) [inline web/app/loading.tsx:3]: Where is this function used?

File: web/app/proposal/[id]/results/loading.tsx [hunk 2]
```diff
@@ -0,0 +1,111 @@
+'use client';
```
Comment (robinbb) [inline web/app/proposal/[id]/results/loading.tsx:1]: How do you intend for the function(s) in this file (e.g. PageLoading()) to be a part of the web app? From where are the functions called?

File: web/app/proposal/[id]/results/page.tsx [hunk 3]
```diff
@@ -11,7 +11,6 @@ interface PageParams {
 
 const Page = async ({ params }: { params: PageParams }) => {
   const proposal = await getProposalResults(params.id);
-
```
Comment (robinbb) [inline web/app/proposal/[id]/results/page.tsx:14]: Stray whitespace change?

File: web/app/styles.css [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    transform: rotate(0deg);
+  }
+  100% {
+    transform: rotate(360deg);
+  }
+} 
+@keyframes animloader {
+    50% {
+      transform: scale(1) translate(-50%, -50%);
+}
```
Comment (robinbb) [inline web/app/styles.css:68]: Indentation inconsistency, here. Not the most important thing, of course.

File: web/app/proposal/[id]/results/page.tsx (hunk 3 above)
Comment (robinbb) [reply web/app/proposal/[id]/results/page.tsx:14]: ... or is this where you had intended to inject PageLoading/LoadingDots stuff?

File: web/app/loading.tsx (hunk 1 above)
Comment (griffinclark10) [reply web/app/loading.tsx:3]: In the past when I have used instant loading states (https://nextjs.org/docs/app/building-your-application/routing/loading-ui-and-streaming#instant-loading-states) I have implemented them just like this. Reading the link though, I see that best practices is to wrap each page in a <Suspense> tag and pass the loading function as the fallback. 

I have been dealing with some issues with my mac today which has prevented me from being able to build the server. Will deal with this tomorrow and push the changes.

File: web/app/proposal/[id]/results/loading.tsx (hunk 2 above)
Comment (griffinclark10) [reply web/app/proposal/[id]/results/loading.tsx:1]: function is now called in the Suspense component in the page.tsx file

**Comment by robinbb** [comment]
@griffinclark10 I strongly recommend that you `git fetch upstream && git merge upstream/main` (assuming that this repo is `upstream` for you). There have been some fixes on 'main' that will affect your ability to test this using docker and docker-compose.

**Comment by robinbb** [comment]
... when I did that, things began to work for me. Also, you'll have to update the PR to have it merged, in any case.

**Comment by robinbb** [comment]
> @griffinclark10 I strongly recommend that you `git fetch upstream && git merge upstream/main` (assuming that this repo is `upstream` for you). There have been some fixes on 'main' that will affect your ability to test this using docker and docker-compose.

I've done this for you, now.

**Comment by robinbb** [comment]
Deployment with your code is here: https://mina-on-chain-voting-d00kdbc6a-team-granola.vercel.app/proposal/4/results

**Comment by trevorbernard** [comment]
@griffinclark10 Direct linking to the results is stuck loading but linking from the homepage works

**Comment by griffinclark10** [comment]
> @griffinclark10 Direct linking to the results is stuck loading but linking from the homepage works

Hey @trevorbernard, hmm, I can't see that behaviour on my end. Are you talking about using this exact link: https://mina-on-chain-voting-d00kdbc6a-team-granola.vercel.app/proposal/4/results and never getting past the loading screen?

**Comment by trevorbernard** [comment]
Yup that's the behaviour I was seeing

**Comment by griffinclark10** [comment]
> Yup that's the behaviour I was seeing

Thats strange. I checked to see if I could replicate the problem on other browsers and found that on chrome and safari, the vercel link from my previous comment NET::ERR_CERT_COMMON_NAME_INVALID code and am unable to access the website. Is this something you've seen before?

**Pull request closed without merging**